packages/core/src/hooks/data/useUpdate.ts

```diff
@@ -303,78 +314,127 @@ export const useUpdate = <
-                            };
-                        },
-                    );
 
-                    queryClient.setQueriesData(
-                        resourceKeys
-                            .action("one")
-                            .id(id)
-                            .params(preferredMeta ?? {})
-                            .get(preferLegacyKeys),
-                        (previous?: GetListResponse<TData> | null) => {
-                            if (!previous) {
-                                return null;
-                            }
-
-                            return {
-                                ...previous,
-                                data: {
-                                    ...previous.data,
-                                    ...values,
-                                },
-                            };
-                        },
-                    );
+                                const data = previous.data.map(
+                                    (record: TData) => {
+                                        if (
+                                            record.id?.toString() ===
+                                            id?.toString()
+                                        ) {
+                                            record = {
+                                                id,
+                                                ...record,
+                                                ...values,
+                                            } as unknown as TData;
+                                        }
+                                        return record;
+                                    },
+                                );
+                                return {
+                                    ...previous,
+                                    data,
+                                };
+                            },
+                        );
+                    }
+
+                    if (queryCacheUpdateMap.detail) {
+                        queryClient.setQueriesData(
+                            resourceKeys
+                                .action("one")
+                                .id(id)
+                                .params(preferredMeta ?? {})
+                                .get(preferLegacyKeys),
+                            (previous?: GetListResponse<TData> | null) => {
+                                if (!previous) {
+                                    return null;
+                                }
+
+                                if (
+                                    typeof queryCacheUpdateMap.detail ===
+                                    "function"
+                                ) {
+                                    return queryCacheUpdateMap.detail(
+                                        previous,
+                                        values,
+                                    );
```

Comment:
```suggestion
                                    return queryCacheUpdateMap.detail(
                                        previous,
                                        values,
                                        id
                                    );
```
We can also return `id`.